pkg/scheduler/watchers/membercluster/controller.go

```diff
@@ -192,16 +192,15 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 			// Capture label changes.
 			//
-			// Note that the controller runs only when label changes happen on joined clusters.
 			clusterKObj := klog.KObj(newCluster)
-			if newCluster.Spec.State == clusterv1beta1.ClusterStateJoin && !reflect.DeepEqual(oldCluster.Labels, newCluster.Labels) {
-				klog.V(2).InfoS("An member cluster label change has been detected", "memberCluster", clusterKObj)
+			// The cluster is being deleted.
+			if oldCluster.GetDeletionTimestamp().IsZero() && !newCluster.GetDeletionTimestamp().IsZero() {
+				klog.V(2).InfoS("A member cluster is leaving the fleet", "memberCluster", clusterKObj)
 				return true
 			}
-
-			// The cluster has left.
-			if oldCluster.Spec.State == clusterv1beta1.ClusterStateJoin && newCluster.Spec.State == clusterv1beta1.ClusterStateLeave {
-				klog.V(2).InfoS("A member cluster has left the fleet", "memberCluster", clusterKObj)
+			// Note that the controller runs only when label changes happen on joined clusters.
+			if !reflect.DeepEqual(oldCluster.Labels, newCluster.Labels) {
```

Comment:
```suggestion
			if newCluster.GetDeletionTimestamp().IsZero() && !reflect.DeepEqual(oldCluster.Labels, newCluster.Labels) {
```

i tested that the labels still can be updated in a cr with deletiontimesamp.

Reply:
confirmed with Chen and scheduler will check the deletionTimestamp again.